src/extension/ipc.cc

```diff
@@ -137,8 +137,7 @@ bool sapi_ipc_reply (const sapi_ipc_result_t* result) {
   }
 
   // if retained, then then caller must eventually call `sapi_context_release()`
-  if (!context->retained) {
-    context->release();
+  if (context->release()) {
```

Comment:
Ah wait, this doesn't seem right

Reply:
For this to work, we have to allocate the route context for each message. See my latest commit.